src/Evaluator.ml

```diff
@@ -81,6 +86,19 @@ let rec eval_open gamma Location.{ data = expr; _ } =
       match pv with
       | Val.Pair (_v1, v2) -> return v2
       | _ -> Result.fail "snd is stuck" )
+  | IntZ i -> return @@ Val.IntZ i
+  | BinOp (op, e1, e2) -> (
+      let%bind lhs = eval_open gamma e1 in
+      let%bind rhs = eval_open gamma e2 in
+      match (lhs, rhs) with
+      | Val.IntZ i1, Val.IntZ i2 -> (
+          match op with
+          | Add -> return @@ Val.IntZ (Nat.add i1 i2)
+          | Sub -> return @@ Val.IntZ (Nat.sub i1 i2)
+          | Mul -> return @@ Val.IntZ (Nat.mul i1 i2)
+          | Div -> return @@ Val.IntZ (Nat.div i1 i2) )
```

Comment:
Let's factor this out as a helper function `interp_binop = function | Add -> Nat.add | Sub -> ...` which interprets a binary operation representation as its corresponding function.
Then this will look like `return @@ Val.Nat (interp_binop op i1 i2)`.

Also, you need to catch the zero division exception here and turn it into `Result.Error`.